test/pyproject.toml

```diff
@@ -0,0 +1,84 @@
+# This project was generated using template: https://github.com/beeware/briefcase-template and branch: v0.3.12
+[tool.briefcase]
+project_name = "Toga test"
+bundle = "org.beeware"
+version = "0.0.1"
+url = "https://beeware.org"
+license = "BSD license"
+author = 'Tiberius Yak'
+author_email = "tiberius@beeware.org"
+
+[tool.briefcase.app.toga-test]
+formal_name = "Toga test"
+description = "Toga test"
+icon = "src/toga_test/resources/toga_test"
+sources = [
+    'src/toga_test',
+    'src/tests',
+]
+requires = [
+    '../core',
+    '../dummy',
+    'pytest==7.2.0',
+    'pytest-asyncio==0.20.2',
+]
+
+
+[tool.briefcase.app.toga-test.macOS]
+requires = [
+    '../cocoa',
+    'std-nslog~=1.0.0'
+]
+
+[tool.briefcase.app.toga-test.linux]
+requires = [
+    '../gtk',
+]
+
+[tool.briefcase.app.toga-test.linux.appimage]
+system_requires = [
+    'gir1.2-webkit-3.0',
+    'libcairo2-dev',
+    'libgirepository1.0-dev',
+    'libgtk-3-dev',
+    'libpango1.0-dev',
+    'librsvg2-dev',
+    'libwebkitgtk-3.0-0',
+]
+linuxdeploy_plugins = [
+    'DEPLOY_GTK_VERSION=3 gtk',
+]
+
+[tool.briefcase.app.toga-test.linux.flatpak]
+flatpak_runtime = 'org.gnome.Platform'
+flatpak_runtime_version = '42'
+flatpak_sdk = 'org.gnome.Sdk'
+
+[tool.briefcase.app.toga-test.windows]
+sources = [
+    '../winforms/test_probes',
+]
```

Comment:
Once we get the existing backend tests working within the test app, and once https://github.com/beeware/briefcase/issues/975 is fixed, this can be merged into the existing directory `../winforms/tests`.